### DISCUSSION TITLE
Is it possible to autogenerate a TypedDict from function arguments?

### DISCUSSION THREAD

**Discussion by FeryET** [python/typing#1115] [Q&A]
Hi.

Normally we define our config objects similar to this:
```
class A:
    def __init__(self, x: int, y: int):
        self.x = x
        self.y = y
AConfig(TypedDict):
    x: int
    y: int

Class Container:
     def __init__(a_config: AConfig):
         # do some logic like saving configs and etc
         self.a = A(**a_config)
```
Is there anyway to auto generate `AConfig` from `A.__init__` signature?

**Comment by erictraut** [comment]
There's no mechanism in the current Python type system to auto-generate this. If it's something you need to do often for many functions, you could write a small script to generate it.

You could also change the constructor to accept the TypedDict as a parameter directly.

```python
AConfig(TypedDict):
    x: int
    y: int

class A:
    def __init__(self, config: AConfig):
        self.x = config["x"]
        self.y = config["y"]

Class Container:
     def __init__(a_config: AConfig):
         # do some logic like saving configs and etc
         self.a = A(a_config)
```

**Reply by FeryET** [reply]
> There's no mechanism in the current Python type system to auto-generate this. If it's something you need to do often for many functions, you could write a small script to generate it.
> 
> You could also change the constructor to accept the TypedDict as a parameter directly.
> 
> ```python
> AConfig(TypedDict):
>     x: int
>     y: int
> 
> class A:
>     def __init__(self, config: AConfig):
>         self.x = config["x"]
>         self.y = config["y"]
> 
> Class Container:
>      def __init__(a_config: AConfig):
>          # do some logic like saving configs and etc
>          self.a = A(a_config)
> ```

Hi. 

Thanks for your reply. The problem with this solution is that in many cases IDEs are usually less capable of tracking what happens exactly inside a dict to later come up with suggestions and etc for input names and then there is the problem of the parameters not being that clear in the constructor of your object. 

Writing an extension to typehint that can automatically generate and track changes of this TypedDict instance seems like a better idea, though I have little experience in this regard, what are some good examples to start learning from?
